cmd/router/main.go

```diff
@@ -0,0 +1,248 @@
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"k8s.io/client-go/util/jsonpath"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"math/rand"
+
+	"github.com/kserve/kserve/pkg/apis/serving/v1alpha1"
+	flag "github.com/spf13/pflag"
+)
+
+var log = logf.Log.WithName("InferenceGraphRouter")
+
+func callService(serviceUrl string, input []byte, res chan<- []byte) error {
+	resp, err := http.Post(serviceUrl, "application/json", bytes.NewBuffer(input))
+	if err != nil {
+		log.Error(err, "An error has occurred for service %s", serviceUrl)
+		return err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Error(err, "error while reading the response")
+		return err
+	}
+	res <- body
+	return nil
+}
+
+func pickupRoute(routes []v1alpha1.InferenceRoute) *v1alpha1.InferenceRoute {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	//generate num [0,100)
+	point := r.Intn(99)
+	start, end := 0, 0
+
+	for _, route := range routes {
+		start += end
+		end += int(*route.Weight)
+		if point >= start && point < end {
+			return &route
+		}
+	}
+	return nil
+}
+
+//Input is a struct that can be parse by jsonpath
+type Input struct {
+	Items []interface{} `json:"items"`
+}
+
+func convertInput(input []byte) (interface{}, error) {
+	var inputData interface{}
+	if err := json.Unmarshal(input, &inputData); err != nil {
+		return nil, err
+	}
+	var data Input
+	data.Items = append(data.Items, inputData)
+	return data, nil
+}
+
+func convertCondition(origin string) string {
+	//remove whitespaces
+	str := strings.Replace(origin, " ", "", -1)
+	//remove {}
+	str = str[1 : len(str)-1]
+	return fmt.Sprintf("{@.items[?(%s)]}", str[strings.Index(str, "."):])
+}
+
+func pickupRouteByCondition(input []byte, routes []v1alpha1.InferenceRoute) *v1alpha1.InferenceRoute {
+	//convert input to Input
+	data, err := convertInput(input)
+	if err != nil {
+		log.Error(err, "converInput failed.")
+		return nil
+	}
+	for _, route := range routes {
+		//new jsonpath object
+		j := jsonpath.New("Parser")
+		//j.AllowMissingKeys(true)
+		cond := convertCondition(route.Condition)
+		if err := j.Parse(cond); err != nil {
+			log.Error(err, "jsonpath.Parse failed")
+			continue
+		}
+		buf := new(bytes.Buffer)
+		if err := j.Execute(buf, data); err != nil {
+			log.Error(err, "jsonpath.Execute failed")
+		}
+		if buf.Len() > 0 { // find the target
+			return &route
+		}
+	}
+	return nil
+}
+
+func timeTrack(start time.Time, name string) {
+	elapsed := time.Since(start)
+	log.Info("elapsed time", "node", name, "time", elapsed)
+}
+
+func routeStep(nodeName string, currentStep v1alpha1.InferenceRouter, graph v1alpha1.InferenceGraphSpec, input []byte, res chan<- []byte) error {
+	log.Info("current step", "nodeName", nodeName, "URL", currentStep.Routes[0].ServiceUrl)
+	defer timeTrack(time.Now(), nodeName)
+	response := map[string]interface{}{}
+	//For splitter and ABNTest call virtual service
```

Comment:
nit: can you add a bit of "why" to this comment? like - why is splitter different here?